src/main/java/dk/alexandra/fresco/lib/helper/builder/AdvancedNumericBuilder.java

```diff
@@ -0,0 +1,28 @@
+package dk.alexandra.fresco.lib.helper.builder;
+
+import dk.alexandra.fresco.framework.ProtocolProducer;
+import dk.alexandra.fresco.framework.value.SInt;
+import dk.alexandra.fresco.framework.value.SIntFactory;
+import dk.alexandra.fresco.lib.math.integer.division.DivisionFactory;
+
+public class AdvancedNumericBuilder extends AbstractProtocolBuilder {
+
+    private DivisionFactory divisionFactory;
+    private SIntFactory intFactory;
+
+    public AdvancedNumericBuilder(DivisionFactory divisionFactory, SIntFactory intFactory) {
+        this.divisionFactory = divisionFactory;
+        this.intFactory = intFactory;
+    }
+
+    public SInt div(SInt p, int maxBitLengthP, SInt q, int maxBitLengthQ) {
```

Comment:
Oh yea.. two things: javadoc missing. One does not know if p/q or q/p. And maybe call them divident and divisor. 
And second thing: We would like methods for all the possible functionality that the division factory provides. So also division by known divisor and division where the remainder is returned as well etc.